src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Extension.cs

```diff
@@ -0,0 +1,82 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols
+{
+    internal partial class SourceNamedTypeSymbol
+    {
+        private ImmutableArray<ParameterSymbol> _lazyExtensionParameters;
```

Comment:
>_lazyExtensionParameters [](http://example.com/codeflow?start=48&length=24)

I think we should have at most one parameter

Reply:
FWIW, I took a stab at not parsing __arglist, so we could have a non-nullable parameter here, but it snowballed a bit, so I ended up keeping the parser as-is.